### PR TITLE
Issue #795 - Add built-in support for CARIN BB IG

### PR DESCRIPTION
**Pull request by JohnTimm**
Signed-off-by: John T.E. Timm <johntimm@us.ibm.com>

### CODE REVIEW

File: docs/src/pages/guides/FHIRValidationGuide.md [hunk 1]
```diff
@@ -139,8 +139,9 @@ For more information, please see: [https://confluence.hl7.org/display/FHIR/NPM+P
 
 The IBM FHIR server has built-in support for the following:
 
-- Profiles defined in the base FHIR specification STU4 (v4.0.1)
-- US Core STU3 (v3.1.0)
+- Profiles defined in the base FHIR specification STU4 (v4.0.1) [http://hl7.org/fhir/profilelist.html](http://hl7.org/fhir/profilelist.html)
+- US Core STU3 (v3.1.0) [https://www.hl7.org/fhir/us/core/index.html](https://www.hl7.org/fhir/us/core/index.html)
+- Consumer-Directed Payer Data Exchange (CARIN Blue Button) STU1 (v0.1.0) [http://hl7.org/fhir/us/carin-bb/2020Feb/](http://hl7.org/fhir/us/carin-bb/2020Feb/)
```
Comment (lmsurpre) [inline docs/src/pages/guides/FHIRValidationGuide.md]: I'm not sure we can call this "STU1" yet.  it was balloted for STU1 but the link there is the ballot version, not an approved STU1 version
```suggestion
- Consumer-Directed Payer Data Exchange (CARIN Blue Button) v0.1.0 [http://hl7.org/fhir/us/carin-bb/2020Feb/](http://hl7.org/fhir/us/carin-bb/2020Feb/)
```

File: fhir-ig-carin-bb/src/test/java/com/ibm/fhir/ig/carin/bb/test/BBResourceProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+package com.ibm.fhir.ig.carin.bb.test;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import com.ibm.fhir.ig.carin.bb.BBResourceProvider;
+import com.ibm.fhir.registry.spi.FHIRRegistryResourceProvider;
+
+public class BBResourceProviderTest {
```
Comment (lmsurpre) [inline fhir-ig-carin-bb/src/test/java/com/ibm/fhir/ig/carin/bb/test/BBResourceProviderTest.java]: maybe add a test to loop through all the resources and validate them?

File: docs/src/pages/guides/FHIRValidationGuide.md (hunk 1 above)
Comment (JohnTimm) [reply docs/src/pages/guides/FHIRValidationGuide.md]: I think we can call it STU1 Ballot 1

**Comment by prb112** [comment]
Just a thought.. maybe you could keep these out of the main build?  Treat them as a fhir-examples like project?  maybe the same approach with core?

File: docs/src/pages/guides/FHIRValidationGuide.md (hunk 1 above)
Comment (JohnTimm) [reply docs/src/pages/guides/FHIRValidationGuide.md]: Updated name(s) used in doc

File: fhir-ig-carin-bb/src/test/java/com/ibm/fhir/ig/carin/bb/test/BBResourceProviderTest.java (hunk 2 above)
Comment (JohnTimm) [reply fhir-ig-carin-bb/src/test/java/com/ibm/fhir/ig/carin/bb/test/BBResourceProviderTest.java]: Added validation to unit test in US Core but left it out of CARIN BB (the validation in IndexGenerator is sufficient per our conversation).

**Pull request merged**